File: src/types/group.ts
export interface Group {
  id: number;
  name: string;
  description?: string;
  avatar?: string;
  createdById: number;
  createdByUsername: string;
  createdAt: string;
  isMutedForAll: boolean;
  memberCount: number;
  members: GroupMember[];
}

export interface GroupMember {
  id: number;
  userId: number;
  username: string;
  avatar?: string;
  isAdmin: boolean;
  isMuted: boolean;
  isOnline: boolean;
  joinedAt: string;
}

export interface GroupMessage {
  id: number;
  groupId: number;
  senderId: number;
  senderUsername: string;
  senderAvatar?: string;
  content: string;
  isSystemMessage: boolean;
  sentAt: string;
}

File: BackNtier/Entity/TableModel/Group.cs
using Core.Entities;
using Entity.TableModel;
using System;
using System.Collections.Generic;

namespace Entity.TableModel
{
    public class Group : BaseEntity
    {
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string? Avatar { get; set; }
        public int CreatedById { get; set; }
        public bool IsMutedForAll { get; set; } = false;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        public User CreatedBy { get; set; } = null!;
        public ICollection<GroupMember> Members { get; set; } = new List<GroupMember>();
        public ICollection<GroupMessage> Messages { get; set; } = new List<GroupMessage>();
    }
}
